openedx/core/djangoapps/programs/tasks/v1/tests/test_tasks.py

```diff
@@ -165,12 +165,20 @@ def test_award_program_certificate(self):
         """
         test_username = 'test-username'
         test_client = EdxRestApiClient('http://test-server', jwt='test-token')
+
         httpretty.register_uri(
             httpretty.POST,
             'http://test-server/user_credentials/',
         )
+
         tasks.award_program_certificate(test_client, test_username, 123)
-        self.assertEqual(httpretty.last_request().body, json.dumps({'program_id': 123, 'username': test_username}))
+
+        expected_body = {
+            'username': test_username,
+            'credential': {'program_id': 123},
+            'attributes': []
+        }
+        self.assertEqual(httpretty.last_request().body, json.dumps(expected_body))
```

Comment:
This question is way beyond the scope of this PR, but I just thought about it: does `json.dumps` come with any guarantees about the ordering of key/value pairs in its output? To put it another (more actionable) way, why are we asserting that `payload == dumps(expected)` instead of `loads(payload) == expected`?